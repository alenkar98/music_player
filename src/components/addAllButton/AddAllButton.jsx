import { IoAdd } from 'react-icons/io5';
import './addAllButton.css';
import { useDispatch, useSelector } from 'react-redux';
import { addAllSongsToQueue, clearQueue } from '../../redux/action/songAction';

export default function AddAllButton() {
  const dispatch = useDispatch();
  const queue = useSelector((state) => state.song.queue);
  const handleAddAllButton = () => {
    if (queue.length > 1) {
      dispatch(clearQueue());
      console.log('All songs cleared from queue');
    } else {
      console.log('All songs added to queue');
      dispatch(addAllSongsToQueue());
    }
  };

  return (
    <button className="addAll" onClick={handleAddAllButton}>
      <IoAdd style={{ display: queue.length > 1 ? 'none' : 'inline-block' }} />
      <span>{queue.length > 1 ? 'Clear All' : 'Add All'}</span>
    </button>
  );
}
