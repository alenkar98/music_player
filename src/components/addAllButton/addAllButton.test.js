import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import AddAllButton from './AddAllButton';

const mockStore = configureStore([]);

test('AddAllButton component renders correctly', () => {
  const initialState = {
    song: {
      queue: [], // Adjust the state based on your actual Redux state structure
    },
  };

  const store = mockStore(initialState);

  const component = renderer.create(
    <Provider store={store}>
      <AddAllButton />
    </Provider>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
