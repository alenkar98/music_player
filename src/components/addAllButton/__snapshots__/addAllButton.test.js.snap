// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAllButton component renders correctly 1`] = `
<button
  className="addAll"
  onClick={[Function]}
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
        "display": "inline-block",
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 112v288m144-144H112"
      fill="none"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="32"
    />
  </svg>
  <span>
    Add All
  </span>
</button>
`;
